homework-submission/src/App.js

```diff
@@ -0,0 +1,206 @@
+    this.setState({ todos: newTodos });
+  };
+
+  handleEdit = itemID => {
+    const { actions, todos } = this.state;
+    let actionsCopy = { ...actions, editClicked: true };
+    actionsCopy.updatedTodo = { ...actionsCopy.updatedTodo, itemID };
+
+    const todo = todos.filter(todo => todo.id === itemID)[0];
+    const description = todo.description;
+    const deadline = todo.deadline;
+    actionsCopy.updatedTodo = {
+      ...actionsCopy.updatedTodo,
+      description,
+      deadline
+    };
+
+    // ✓
+    // console.log(this.state.actions.updatedTodo, actionsCopy.updatedTodo);
+    this.setState({ actions: actionsCopy });
+  };
+
+  handleCancel = () => {
+    const actionsCopy = {
+      ...this.state.actions,
+      editClicked: false
+    };
+    actionsCopy.updatedTodo = {
+      ...actionsCopy.updatedTodo,
+      description: "",
+      deadline: "",
+      itemID: ""
+    };
+
+    // ✓
+    // console.log(this.state.actions, actionsCopy);
+    this.setState({ actions: actionsCopy });
+  };
+
+  handleUpdateDescription = e => {
+    const { actions } = this.state;
+    let actionsCopy = { ...actions };
+    actionsCopy.updatedTodo = {
+      ...actionsCopy.updatedTodo,
+      description: e.target.value
+    };
+
+    // ✓
+    // console.log(this.state.actions, actionsCopy);
+    this.setState({ actions: actionsCopy });
+  };
+
+  handleUpdateDeadline = e => {
+    const { actions } = this.state;
+    let actionsCopy = { ...actions };
+    actionsCopy.updatedTodo = {
+      ...actionsCopy.updatedTodo,
+      deadline: e.target.value
+    };
+
+    // ✓
+    // console.log(this.state.actions, actionsCopy);
+    this.setState({ actions: actionsCopy });
+  };
+
+  handleUpdate = itemID => {
+    const { todos, actions } = this.state;
+    let actionsCopy = { ...actions };
+
+    const todosCopy = todos.map(todo => {
+      if (todo.id === itemID) {
+        todo = {
+          ...todo,
+          description: actions.updatedTodo.description,
+          deadline: actions.updatedTodo.deadline
+        };
+        actionsCopy = { ...actionsCopy, editClicked: false };
+
+        actionsCopy.updatedTodo = {
+          ...actionsCopy.updatedTodo,
+          description: "",
+          deadline: "",
+          itemID: ""
+        };
+      }
+      return todo;
+    });
+
+    // ✓
+    // console.log(this.state.todos, todosCopy);
+    this.setState({ actions: actionsCopy, todos: todosCopy });
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <List
+          dataModel={this.state.todos}
+          handleCheck={this.handleCheck}
+          handleEdit={this.handleEdit}
+          handleRemove={this.handleRemove}
+          handleUpdate={this.handleUpdate}
+          actions={this.state.actions}
+          handleUpdateDescription={this.handleUpdateDescription}
+          handleUpdateDeadline={this.handleUpdateDeadline}
+          handleCancel={this.handleCancel}
+        />
```

Comment:
To be honest I was saying that to myself but I couldn't think of a better way to to that without a state management tool

Reply:
this code is fine, it does what the assignment asks, and you should be proud of yourself for realising it can be improved!
If you want to watch a fun video on how to make things simpler in React, I highly suggest [Simply React from Kent C. Dodds](https://www.youtube.com/watch?v=AiJ8tRRH0f8)

Reply:
You're a saint 
Thank you for the video 👐